src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/Program.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+using System.ComponentModel;
+
+var builder = Host.CreateApplicationBuilder(args);
+
+if (args.Length == 0 || args[0] != "start-mcp")
+{
+    Console.Error.WriteLine("Error: invalid command. Use the 'start-mcp' command like argument to start the MCP server.");
+    return 1;
+}
+
+var maxNumberEnv = Environment.GetEnvironmentVariable("MAX_RANDOM_NUMBER");
+if (!int.TryParse(maxNumberEnv, out var maxNumber) || maxNumber <= 0)
+{
+    Console.Error.WriteLine("Error: you must set the MAX_RANDOM_NUMBER environment variable to a positive integer.");
+    return 1;
+}
```

Comment:
For simplicity you could just default it, e.g.
```C#
var maxNumberEnv = Environment.GetEnvironmentVariable("MAX_RANDOM_NUMBER");
if (!int.TryParse(maxNumberEnv, out var maxNumber) || maxNumber is <= 0 or int.MaxValue)
{
    maxNumber = int.MaxValue - 1;
}
```

Reply:
I chose to force the ENV to simulate a required env case, such as an API key. VS Code has a reasonable error experience if MCP fails to start.

Reply:
I'm going to close this since I want authors to see and potentially bump into this error experience when testing.